pkg/deploy/tls/certificates.go

```diff
@@ -0,0 +1,175 @@
+
+	"github.com/eclipse-che/che-operator/pkg/deploy"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+const (
+	injector = "config.openshift.io/inject-trusted-cabundle"
+	// CheCACertsConfigMapLabelKey is the label value which marks config map with additional CA certificates
+	CheCACertsConfigMapLabelValue = "ca-bundle"
+	// CheAllCACertsConfigMapName is the name of config map which contains all additional trusted by Che TLS CA certificates
+	CheAllCACertsConfigMapName = "ca-certs-merged"
+	// CheMergedCAConfigMapRevisionsAnnotationKey is annotation name which holds versions of included config maps in format: cm-name1=ver1,cm-name2=ver2
+	CheMergedCAConfigMapRevisionsAnnotationKey = "che.eclipse.org/included-configmaps"
+
+	// Local constants
+	// labelEqualSign consyant is used as a replacement for '=' symbol in labels because '=' is not allowed there
+	labelEqualSign = "-"
+	// labelCommaSign consyant is used as a replacement for ',' symbol in labels because ',' is not allowed there
+	labelCommaSign = "."
+)
+
+type CertificatesReconciler struct {
+	deploy.Reconcilable
+}
+
+func NewCertificatesReconciler() *CertificatesReconciler {
+	return &CertificatesReconciler{}
+}
+
+func (c *CertificatesReconciler) Reconcile(ctx *deploy.DeployContext) (reconcile.Result, bool, error) {
+	if ctx.Proxy.TrustedCAMapName != "" {
+		if ctx.CheCluster.Spec.Server.ServerTrustStoreConfigMapName == "" {
+			ctx.CheCluster.Spec.Server.ServerTrustStoreConfigMapName = deploy.DefaultServerTrustStoreConfigMapName()
+			if err := deploy.UpdateCheCRSpec(ctx, "ServerTrustStoreConfigMapName", deploy.DefaultServerTrustStoreConfigMapName()); err != nil {
+				return reconcile.Result{}, false, err
+			}
+		}
+
+		result, done, err := c.syncTrustStoreConfigMapToCluster(ctx)
+		if !done {
+			return result, done, err
+		}
+	}
+
+	return c.syncAdditionalCACertsConfigMapToCluster(ctx)
+}
+
+func (c *CertificatesReconciler) Finalize(ctx *deploy.DeployContext) error {
+	return nil
+}
+
+func (c *CertificatesReconciler) syncTrustStoreConfigMapToCluster(ctx *deploy.DeployContext) (reconcile.Result, bool, error) {
+	name := ctx.CheCluster.Spec.Server.ServerTrustStoreConfigMapName
+	configMapSpec := deploy.GetConfigMapSpec(ctx, name, map[string]string{}, deploy.DefaultCheFlavor(ctx.CheCluster))
+
+	// OpenShift will automatically injects all certs into the configmap
+	configMapSpec.ObjectMeta.Labels[injector] = "true"
+	configMapSpec.ObjectMeta.Labels[deploy.KubernetesPartOfLabelKey] = deploy.CheEclipseOrg
+	configMapSpec.ObjectMeta.Labels[deploy.KubernetesComponentLabelKey] = CheCACertsConfigMapLabelValue
+
+	actual := &corev1.ConfigMap{}
+	exists, err := deploy.GetNamespacedObject(ctx, name, actual)
+	if err != nil {
+		return reconcile.Result{}, false, err
+	}
+
+	if !exists {
+		// We have to create an empty config map with the specific labels
```

Comment:
1. Dangerous. If it already exists and not merked with part-of label it will fall into infinite loop
2. Why just not to skip it if the Config Map doesn't exist?

Reply:
we need empty configmap to inject labels to allow OpenShift inject certificates into